src/mixin.js

```diff
@@ -127,8 +125,6 @@ export default {
         self._localeWatcher()
         delete self._localeWatcher
       }
-
-      self._i18n = null
```

Comment:
In your PR, If you do not set the instance, `_i18n` field to null, memory leak might occur.

As the workaround for memory leaks, In `beforeCreate` hook, I think you need to listen to `hook:destroyed` with `$once`, and handle that in it.

e.g. 
```js
beforeCreate (): void {
  // ...
  this.$once('hook:destroyed', () => {
    this._i18n = null
  })
}
...
```

In about `hook:destroyed`, `hook:xxx` is an internal API of Vue.
you can understand about it.
https://github.com/vuejs/vue/blob/87bad80f0cb9a30b95d9410120ff6e3e2022a723/test/unit/features/options/lifecycle.spec.js#L314

Reply:
Or, You can use `destroyed` hook too.

Reply:
I just tested with the hook destroyed. In our case the component is removed from the DOM because another route is visited, but a method of the components still waits from a server response that can take some time >10sec. The server response is after the components destroyed() hook is called.

Reply:
@kazupon the goal of this pull request is to have access to `this._i18n` after the component has been destroyed, as promises (for example AJAX requests) might still be pending. When these promises resolve or reject, `this._i18n` might be used, for example to emit an alert or error, but maybe also to update the local state. Alternatives I see:

- every promise should be canceled when the component is destroyed. This is cumbersome and in quite some cases not even desired, eg. in order to keep the local application state in sync with the server, I'd rather know the outcome of an AJAX request which POST's/PUT's/DELETE's than that I'd cancel it midway
- after every `await`, it should be checked whether the component still exists. That feels like an anti-pattern.

I would therefore argue, that there are valid reasons not to destroy `this._i18n` when the component is destroyed, but only when we are sure that it will not be used any more. You state that not setting `self._i18n = null` might introduce a memory leak. Could you elaborate on that? If there are no references to it any more, won't it be removed by the garbage collector?